Improve clipboard accessibility, refs #13535

mInnes-archives opened the pull request:
PR adds accessibility fixes and enhancements for screen readers to the clipboard module, as well as a small clipboard related improvement for table view in repository search. Improvements include:

- **Clipboard ID form no longer sharing label association with user password form.** Depending on the SR you use, this can cause behaviour such as multiple labels being heard for one form, or none at all.
- **Add legend to 'Load clipboard' fieldset.** I've opted for an 'sr-only' legend to avoid adding what could be seen as superfluous information for sighted users.
- **Add aria-label to help button.**
- **Add text to clipboard column header for table view in repository search.** Without header text this column label was either not identified or identified as 'column 5'. I've opted for 'sr-only' to keep the look of the table unchanged.

Testing utilised Orca and NVDA screen readers.

jraddaoui commented on apps/qubit/modules/clipboard/actions/loadAction.class.php:
```diff
@@ -98,9 +98,9 @@ public function execute($request)
-                $this->form->setWidget('password', new sfWidgetFormInput());
+            case 'clipboardPassword':
+                $this->form->setValidator('clipboardPassword', new sfValidatorString(['required' => true]));
+                $this->form->setWidget('clipboardPassword', new sfWidgetFormInput());
```
Nice! I believe this was also throwing a warning in the browser's dev tools.

jraddaoui reviewed (approved):
Great! Thanks @mInnes-archives!

Please, see https://github.com/artefactual/atom/pull/1312#issuecomment-867393714 and use the same ticket for the reference.

jraddaoui commented on apps/qubit/modules/repository/templates/_browseTableView.php:
```diff
@@ -42,6 +42,7 @@
       </th>
 
       <th style="width 3%">
+        <span class="sr-only"><?php echo __('Clipboard'); ?></span>
```
I didn't know this was available in Bootstrap 2!

mInnes-archives replied:
Apparently it's also part of Font Awesome.

mInnes-archives committed:
Improve clipboard accessibility, refs #13535

Squashed commit of the following:

commit 6a60f4504ad79b798cd6897f3804cba832f6b1a0
Author: Matt Innes <matthew.f.innes@gmail.com>
Date:   Tue Jun 15 23:41:17 2021 -0400

    Add SR-only header to clipboard column

commit b3a6600dd881b4bc3228c8b1afe89b94fcc40630
Author: Matt Innes <matthew.f.innes@gmail.com>
Date:   Mon Jun 14 22:14:35 2021 -0400

    Add screen reader only legend to fieldset

commit 47cdf4e6ba26839e919a15a8102fa4cb0bd963b7
Author: Matt Innes <matthew.f.innes@gmail.com>
Date:   Mon Jun 14 22:13:48 2021 -0400

    Remove duplicate form label association

    - Label for 'password' was associated with both user login
      password and clipboard ID forms causing problems for
      screen readers

commit 0335202bc63cf55808b5f0172ce36be264c7b2a1
Author: Matt Innes <matthew.f.innes@gmail.com>
Date:   Mon Jun 14 21:48:43 2021 -0400

    Add accessible label to help button

mInnes-archives commented:
This one's good to go.

jraddaoui commented:
Thanks @mInnes-archives!
